pkg/ddevapp/ddevapp.go

```diff
@@ -906,18 +906,21 @@ func (app *DdevApp) RestoreSnapshot(snapshotName string) error {
 	if !fileutil.FileExists(hostSnapshotDir) {
 		return fmt.Errorf("Failed to find a snapshot in %s", hostSnapshotDir)
 	}
-	if !fileutil.FileExists(filepath.Join(hostSnapshotDir, "db_mariadb_version.txt")) {
+
+	// This command returning an error indicates grep has failed to find the value
+	_, _, err := app.Exec("db", "sh", "-c", "mysql -V | grep 10.2")
+	if !fileutil.FileExists(filepath.Join(hostSnapshotDir, "db_mariadb_version.txt")) && err == nil {
 		return fmt.Errorf("snapshot %s is not compatible with this version of ddev and mariadb. Please use the instructions at %s for a workaround to restore it", snapshotDir, "https://ddev.readthedocs.io/en/latest/users/troubleshooting/#old-snapshot")
 	}
 
 	if app.SiteStatus() == SiteRunning || app.SiteStatus() == SiteStopped {
-		err := app.Down(false, false)
+		err = app.Down(false, false)
```

Comment:
I don't think we want this, people don't need an extra snapshot when they're trying to restore one?